pkg/controller/resourcecleanup/resourcecleanup_controller.go

```diff
@@ -218,9 +219,14 @@ func (r *ReconcileResourceCleanup) Cleanup(ctx context.Context, cluster *cluster
 	if len(works.Items) == 1 && works.Items[0].Name == klusterletCRDWorkName {
 		// the manifestWorks are deleted by registration controller.
 		// the agent may be orphaned if the CRD manifestWork is force deleted directly.
-		if works.Items[0].DeletionTimestamp.IsZero() {
+		// so the crd is deleting when the deleting condition of crd work is true
+		// and can force delete the crd work after the crd work is deleting.
+		crdWork := works.Items[0]
+		if crdWork.DeletionTimestamp.IsZero() ||
+			!meta.IsStatusConditionTrue(crdWork.Status.Conditions, workv1.WorkDeleting) {
```

Comment:
I believe `!meta.IsStatusConditionTrue(crdWork.Status.Conditions, workv1.WorkDeleting` is sufficient. What do you think?